### ISSUE TITLE
Help Using content_security_policy_style_nonce

### ISSUE THREAD

**Issue by dalezak** [twitter/secureheaders#328]
I'm struggling to get _WYSIWYG_ editors like [Froala](https://www.froala.com/wysiwyg-editor) to work with `style_src: %w('self')`, all seem to cause an violation with `style-src` due to inline styles. 

_Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self' fonts.gstatic.com". Either the 'unsafe-inline' keyword, a hash ('sha256-iECgl+DQnRhCoJAZvXsuvD/twj/lAfgSGNGuDV8nJs0='), or a nonce ('nonce-...') is required to enable inline execution._

I've logged an [issue with Froala](https://github.com/froala/wysiwyg-editor/issues/1946) but unfortunately no luck there.

I'd rather not enable `unsafe-inline`, but how can I use `sha256-` or `nonce-`? Is it possible to include `content_security_policy_style_nonce` in `SecureHeaders::Configuration.configure`?

I can't seem to find any documentation on how to make `sha256-` or `nonce-` work in a Rails app, in particular with Coffeescript. Any help or direction would be greatly appreciated!

**Comment by oreoshake** [comment]
I have not tried in some time, but restricting inline styles is (was) not practical. If your inline styles are static `style` tags (which I doubt) see the [docs on working with hashes](https://github.com/twitter/secureheaders/blob/master/docs/hashes.md).

I'm not familiar with froala, but wysiwig editors are applying styles at runtime and unless they are dropping static style tags that are known ahead of time (which I doubt), any CSP-based whitelisted will not work. 

If you wanted to use hashes, you'd have to precompute the values ahead of time. If you wanted to use nonces, the froala would have to be nonce-aware (and would be limited to style tags).

In other words, I don't think it will ever work. The risk of allowing inline styles is minimal at worst. 

Depending on how your application is architected, you can limit this risk by only applying `unsafe-inline` to pages with the wysiwyg editor.
